test/unit/core/store/middleware.ts

```diff
@@ -175,10 +182,49 @@ suite('Middleware', ({ expect, spy, stub }) => {
     });
   });
 
+  describe('checkPastPurchaseSkus()', () => {
+    it('should pass action through if not in PAST_PURCHASE_SKU_ACTIONS', () => {
+      const action = { type: 'NOT_ACTUALLY_AN_ACTION' };
+
+      Middleware.checkPastPurchaseSkus(<any>{})(null)(next)(action);
+
+      expect(next).to.be.calledWith(action);
+    });
+
+    it('should pass action through if past purchases present', () => {
+      const action = { type: PAST_PURCHASE_SKU_ACTIONS[0] };
+      const state = 'st';
+      const getState = stub().returns(state);
+      const pastPurchases = stub(Selectors, 'pastPurchases').returns([1]);
+
+      Middleware.checkPastPurchaseSkus(<any>{ store: { getState }})(null)(next)(action);
+
+      expect(next).to.be.calledWithExactly(action);
+      expect(getState).to.be.calledWithExactly();
+      expect(pastPurchases).to.be.calledWithExactly(state);
+    });
+
+    it('should call flux.on if past purchases not present', () => {
+      const action = { type: PAST_PURCHASE_SKU_ACTIONS[0] };
+      const state = 'st';
+      const getState = stub().returns(state);
+      const on = spy();
+      const dispatch = spy();
+      const pastPurchases = stub(Selectors, 'pastPurchases').returns([]);
+
+      Middleware.checkPastPurchaseSkus(<any>{ store: { getState, dispatch }, on })(null)(next)(action);
+      on.getCall(0).args[1]();
```

Comment:
why do we do this here?

Reply:
To call the function passed to `flux.on` and see if it does what expected.